system-setup.py

```diff
@@ -22,39 +22,24 @@ def common_first(self):
         self.install("git jq curl")
         self.run("%s/bin/enable-utf8" % READIES)
 
-        self.install("autoconf libtool")
-
-        if self.os == 'linux':
-            if self.dist == 'fedora':  # also include centos 8
-                self.install("openssl-devel")
-            elif self.platform.is_debian_compat():
-                self.install("libssl-dev")
-            elif self.platform.is_redhat_compat():
-                self.install("openssl-devel")
-            elif self.dist == 'suse':
-                self.install("openssl")
-            elif self.dist == 'arch':
-                self.install("openssl")
-            elif self.dist == 'alpine':
-                self.install("openssl")
-        elif self.os == 'macos':
-            self.install("openssl")
-        elif self.os == 'freebsd':
-            self.install("openssl")
+        self.install("autoconf libtool m4 automake")
+        self.install("openssl")
```

Comment:
@rafie Does all packages familiar with openssl?

Reply:
@OfirMos Yes, this is the assumption here. If we're proven wrong, we either remove it from here and add it in the os-specific method (which is a drag as most of them will be the same), Or we let it run with try-mode (meaning, letting it fail), and do the right installation in the OS that doesn't have openssl package
.

Reply:
@OfirMos But then again, "openssl" is usually not the dev artifacts, just the utilities and system configuration, so most systems will have it. Like, it's simple enough, just like git, curl, etc.